crates/cargo-contract/src/cmd/call.rs

```diff
@@ -143,6 +147,7 @@ impl CallCommand {
             .file(self.extrinsic_cli_opts.file.clone())
             .manifest_path(self.extrinsic_cli_opts.manifest_path.clone())
             .url(self.extrinsic_cli_opts.url.clone())
+            .chain(self.extrinsic_cli_opts.chain.clone())
```

Comment:
Why do we need to pass chain to `extrinsics` crate? I think `extrinisics` crate should just get url as it gets it now. We could keep all the prod chain logic in `cargo-contracts` crate

Reply:
We also want to retain info about what kind of chain it is: `Prod` or `Custom`. So it is easier to encapsulate the logic this way.

Reply:
Why do we need this information in `extrinsics` crate? The upload check is done in `cargo-contract` crate.
In `ExtrinsicOpts` we have url and chain which is kind of data duplication.

Reply:
I agree this is not belong in `extrinsics`